### PR TITLE
add host manager controller

### PR DESCRIPTION
**Pull request by OmriAlster**

### CODE REVIEW

File: src/dto/create-game-request.dto.ts [hunk 1]
```diff
@@ -29,12 +31,16 @@ export class CreateGameRequestDto {
   @Max(30)
   totalRounds: MaxInt<30>;
 
-  @IsUrl()
-  @Matches(SPOTIFY_PLAYLIST_URL_REGEX)
-  @ValidateIf((o) => !o.genre || o.playlistUrl)
-  playlistUrl?: string;
+  @IsString()
+  @Length(22)
```
Comment (orenhta) [inline src/dto/create-game-request.dto.ts:35]: why 22?
do they state that in their api?

File: src/dto/game-settings.dto.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export interface GameSettingsDto {
```
Comment (orenhta) [inline src/dto/game-settings.dto.ts]: why?

File: src/dto/create-game-request.dto.ts [hunk 3]
```diff
@@ -29,12 +31,16 @@ export class CreateGameRequestDto {
...
-  @ValidateIf((o) => !o.playlistUrl || o.genre)
-  genre?: Genre;
+  // @IsUrl()
+  // @Matches(SPOTIFY_PLAYLIST_URL_REGEX)
+  // @ValidateIf((o) => !o.genre || o.playlistUrl)
+  // playlistUrl?: string;
+
+  // @IsEnum(Genre)
+  // @ValidateIf((o) => !o.playlistUrl || o.genre)
+  // genre?: Genre;
```
Comment (orenhta) [inline src/dto/create-game-request.dto.ts]: remove

File: src/functions/as-dtos.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import {
+  SimplifiedPlaylist,
+  Playlist as SpotifyPlaylist,
```
Comment (orenhta) [inline src/functions/as-dtos.ts:3]: why?

File: src/modules/game-manager/game-manager.gateway.ts [hunk 5]
```diff
@@ -55,6 +57,17 @@ export class GameManagerGateway implements OnGatewayDisconnect {
     return gameCreationResponse;
   }
 
+  @SubscribeMessage('prepare-game')
```
Comment (orenhta) [inline src/modules/game-manager/game-manager.gateway.ts]:
```suggestion
  @SubscribeMessage(‘get-available-playlists')
```

File: src/modules/music-api/spotify.repository.ts [hunk 6]
```diff
@@ -30,17 +27,21 @@ export class SpotifyRepository {
...
-    amount: MaxInt<100>,
-    playlistUrl: string,
-  ): Promise<Song[]> {
-    const playlistId = playlistUrl.match(SPOTIFY_PLAYLIST_URL_REGEX)![1];
-
-    return this.getSongsByPlaylistId(amount, playlistId);
+  async getMasterPlaylists() {
+    return await Promise.all(
+      Object.values(this.playlistIdByGenre).map(
+        async (id) => await this.spotify.playlists.getPlaylist(id),
```
Comment (orenhta) [inline src/modules/music-api/spotify.repository.ts]:
```suggestion
        (id) => this.spotify.playlists.getPlaylist(id),
```

Comment (orenhta) [reply src/modules/music-api/spotify.repository.ts]: you do promise all so there is no need for that

File: src/dto/game-settings.dto.ts (hunk 2 above)
Comment (orenhta) [reply src/dto/game-settings.dto.ts]: not needed

File: src/functions/as-dtos.ts [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+  Track,
```
Comment (orenhta) [inline src/functions/as-dtos.ts]: unused. delete

File: src/dto/create-game-request.dto.ts (hunk 1 above)
Comment (OmriAlster) [reply src/dto/create-game-request.dto.ts:35]: yes

**Pull request merged**